src/categorical_algebra/ShapeDiagrams.jl

```diff
@@ -1,10 +1,20 @@
 """ Diagrams of a given shape.
 """
 module ShapeDiagrams
-export Span, Cospan, left, right, apex, base
+export AbstractFunctor, AbstractLaxitor, LaxMonoidalFunctor,
+       Span, Cospan, DecoratedCospan, left, right, apex, base,
+       decorator, decoration, undecorate
 
 using ...Theories: dom, codom
 
+abstract type AbstractFunctor end
```

Comment:
Strictly speaking, stuff about lax monoidal functors doesn't belong in this module, but I think that's OK for now. Once we have a fully generic system for decorated cospans, we can create a separate module for decorated cospans and put everything there.

Reply:
Yeah I agree it doesn't really fit in this module, so we can definitely move it once it becomes sufficiently sophisticated.